src/crypto/CHIPCryptoPALPSA.cpp

```diff
@@ -1309,8 +1309,11 @@ CHIP_ERROR VerifyAttestationCertificateFormat(const ByteSpan & cert, Attestation
                 result = mbedtls_asn1_get_bool(&p, end, &isCA);
                 VerifyOrExit(result == 0 || result == MBEDTLS_ERR_ASN1_UNEXPECTED_TAG, error = CHIP_ERROR_INTERNAL);
 
-                if (p != seqStart + len)
+                // Missing pathLen optional tag will leave pathLen == -1.
+                bool hasPathLen = (p <= (seqStart + len));
```

Comment:
So just to make sure I understand....  we `mbedtls_asn1_get_tag`.  This updates `p` to point right after the tag + length, so now it points to the value, ok.  `len` is the length of the value.  I guess this value is the sequence?

We then mbedtls_asn1_get_bool.  This updates `p` to point to after the boolean's tag+value.

NOw we compare `seqStart + len` to `p`.  But what about the size of the encoded sequence length, the size of the boolean, etc? 

I am assuming I am missing something here in terms of how these APIs work.

Reply:
Yes. These APIs are brutal and very low-level and deal with the pointer. It doesn't help that this is nested parsing of ASN.1 within an octet string. This piece of code here for detecting optional element is still in sequence works only for the particular case of elements within a sequence. I did instrumented tests with a debugger with @emargolis and we validated that we correctly detected the presence/absence of the tag, and the pointers/sizes were sane. I hate this mbedTLS low-level API :(